sunpy/map/maputils.py

```diff
@@ -375,3 +375,27 @@ def on_disk_bounding_coordinates(smap):
     return SkyCoord([np.nanmin(tx), np.nanmax(tx)] * u.arcsec,
                     [np.nanmin(ty), np.nanmax(ty)] * u.arcsec,
                     frame=smap.coordinate_frame)
+
+
+def contains_coordinate(smap, coordinates):
+    """
+    Checks whether a coordinate falls within the bounds of a map.
+
+    Parameters
+    ----------
+    smap : `~sunpy.map.GenericMap`
+        The input map.
+
+    coordinates : `~astropy.coordinates.SkyCoord`
+        The input coordinate.
+
+    Returns
+    -------
+    `~bool`
+        Returns ``True`` if ``coordinates`` falls within the bounds of ``smap``.
+    """
+    # Dimensions of smap
+    xs, ys = smap.dimensions
+    # Converting coordinates to pixels
+    xc, yc = smap.world_to_pixel(coordinates)
+    return xc >= -0.5 and xc <= xs-0.5 and yc >= -0.5 and yc <= ys-0.5
```

Comment:
```suggestion
    return (xc >= -0.5*u.pix) * (xc <= xs - 0.5*u.pix) * (yc >= -0.5*u.pix) * (yc <= ys - 0.5*u.pix)
```
The existing code doesn't work